File: java/easy/268_missing_number.java
package java.easy;

class MissingNumber {
    public int missingNumber(int[] nums) {
        int result = nums.length;
        for (int i = 0; i < nums.length; i++) {
            result = result ^ i ^ nums[i];
        }

        return result;
    }
}

File: go/easy/268_missing_number.go
// https://leetcode.com/problems/missing-number/
package easy

func MissingNumber(nums []int) int {
	n := len(nums)
	total := n * (n + 1) / 2 // https://brilliant.org/wiki/sum-of-n-n2-or-n3/
	sum := 0
	for i := range nums {
		sum += nums[i]
	}
	return total - sum
}
